### PR TITLE
dialects: (comb) Fix extract op

### PR DESCRIPTION
**Pull request by Moxinilian**
The constructor, parser and printer of the comb extract op are incorrect. This PR fixes it and adds a verifier.

### CODE REVIEW

**Comment by webmiche** [comment]
comb is a CIRCT dialect, right? If so, which CIRCT commit are we talking about here?

Or is comb in MLIR? In this case, I would prefer to add some interoperation tests to this PR.

**Comment by Moxinilian** [comment]
@webmiche Comb is CIRCT. We are talking about commit `2e23cda6c2cbedb118b92fab755f1e36d80b13f5`.

**Comment by webmiche** [comment]
Hmm, I think this commit hash is important. Is there a way to include it? Maybe add it at the top of the file?

If what was there before is broken, we can only be happy that the almighty Theo decided to bless us mere humans with this correct code. Tysm!

**Comment by webmiche** [comment]
That's fine, it's better than before, so i am happy :+1:

**Review by webmiche — APPROVED** [review]

File: xdsl/dialects/comb.py [hunk 1]
```diff
@@ -376,30 +383,55 @@ class ExtractOp(IRDLOperation):
     result: OpResult = result_def(IntegerType)
 
     def __init__(
-        self, operand: Operation | SSAValue, low_bit: IntegerAttr[IntegerType]
+        self,
+        operand: Operation | SSAValue,
+        low_bit: IntegerAttr[IntegerType],
+        result: IntegerType,
```
Comment (math-fehr) [inline xdsl/dialects/comb.py:389]:
```suggestion
        result_type: IntegerType,
```

File: xdsl/dialects/comb.py [hunk 2]
```diff
@@ -376,30 +383,55 @@ class ExtractOp(IRDLOperation):
...
     ):
         operand = SSAValue.get(operand)
         return super().__init__(
             attributes={"low_bit": low_bit},
             operands=[operand],
-            result_types=[operand.type],
+            result_types=[result],
```
Comment (math-fehr) [inline xdsl/dialects/comb.py:395]:
```suggestion
            result_types=[result_type],
```

**Review by math-fehr — APPROVED** [review]
Nice!

**Pull request merged**